### PR TITLE
DRIVERS-2789 Add Generated Markdown Index File

### PR DESCRIPTION
**Pull request by blink1073**
After this is merged I can open a ticket to add this repo to our internal docs.

### CODE REVIEW

File: scripts/generate_index.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import os
+from pathlib import Path
+source = Path(__file__).resolve().parent.parent / "source"
+source = source.resolve()
+info = {}
+for p in Path(source).rglob("*"):
+    if p.suffix not in ['.md']:
+        continue
+    relpath = os.path.relpath(p.parent, start=source)
```
Comment (Jibola) [inline scripts/generate_index.py:9]: NIT: We could use `"*.md"` reduce the amount of paths to iterate. Thoughts?
```suggestion
for p in Path(source).rglob("*.md"):
    relpath = os.path.relpath(p.parent, start=source)
```

File: scripts/generate_index.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    if "tests" in relpath:
+        continue
+    if p.name in ['index.md']:
+        continue
+    fpath = relpath + '/' + p.name
+    name = None
+    with p.open() as fid:
+        for line in fid:
+            if line.startswith("# "):
+                name = line.replace('# ', '').strip()
```
Comment (Jibola) [inline scripts/generate_index.py:19]: NIT: Is it guaranteed that all the `.md` files have headings that are hyphen delimited?

**Review by Jibola — APPROVED** [review]
LGTM

File: scripts/generate_index.py (hunk 1 above)
Comment (blink1073) [reply scripts/generate_index.py]: Done

File: scripts/generate_index.py (hunk 2 above)
Comment (blink1073) [reply scripts/generate_index.py:17]: Do you mean whether they start with `# `?  There is a guard for that.  For hyphens, I'm adding them explicitly below.

**Pull request merged**